app/app.go

```diff
@@ -191,9 +191,24 @@ func Run(ctx context.Context, conf Config) (err error) {
 	}
 	initStartupMetrics()
 
-	wireMonitoringAPI(life, conf.MonitoringAddr, localEnode)
+	pubkeys, err := eth2PubKeys(lock.Validators)
+	if err != nil {
+		return err
+	}
 
-	if err := wireCoreWorkflow(ctx, life, conf, lock, nodeIdx, tcpNode, p2pKey); err != nil {
+	eth2Cl, beaconAddr, err := newETH2Client(ctx, conf, life, pubkeys)
+	if err != nil {
+		return err
+	}
+
+	peerIDs, err := lock.PeerIDs()
+	if err != nil {
+		return err
+	}
+
+	wireMonitoringAPI(ctx, life, conf.MonitoringAddr, localEnode, tcpNode, eth2Cl.(eth2client.NodeSyncingProvider), peerIDs)
```

Comment:
rather pass in eth2cl directly, and then cast inside and catch and return the error